homeassistant/components/sensor/imap_email_content.py

```diff
@@ -202,12 +206,18 @@ def get_msg_text(email_message):
         message_untyped_text = None
 
         for part in email_message.walk():
+            if not part.get_payload():
+                continue
+
             if part.get_content_type() == 'text/plain':
                 if message_text is None:
                     message_text = part.get_payload()
             elif part.get_content_type() == 'text/html':
                 if message_html is None:
-                    message_html = part.get_payload()
+                    if part['Content-Transfer-Encoding'] == 'base64':
+                        message_html = base64.decodestring(part.get_payload().encode('utf-8'))
```

Comment:
If you use `base64.b64decode` instead of `base64.decodestring` you should be able to skip the encode step.